src/api/pickAppResource.ts

```diff
@@ -3,18 +3,43 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { AzExtTreeItem, ITreeItemPickerContext } from "@microsoft/vscode-azext-utils";
+import { AzExtResourceType, AzExtTreeItem, compatibilityPickAppResourceExperience, ContextValueFilter, getAzExtResourceType, ITreeItemPickerContext } from "@microsoft/vscode-azext-utils";
 import { PickAppResourceOptions } from "@microsoft/vscode-azext-utils/hostapi";
 import { ext } from "../extensionVariables";
-import { SubscriptionTreeItem } from "../tree/SubscriptionTreeItem";
 
 export async function pickAppResource<T extends AzExtTreeItem>(context: ITreeItemPickerContext, options?: PickAppResourceOptions): Promise<T> {
-    const subscription: SubscriptionTreeItem = await ext.appResourceTree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
-    const appResource = await subscription.pickAppResource(context, options);
+    return await compatibilityPickAppResourceExperience<T>(context, ext.v2.resourceGroupsTreeDataProvider, {
+        resourceTypes: convertAppResourceFilterToAzExtResourceType(options?.filter),
+        childItemFilter: convertExpectedChildContextValueToContextValueFilter(options?.expectedChildContextValue)
+    });
+}
+
+function convertExpectedChildContextValueToContextValueFilter(expectedChildContextValue?: PickAppResourceOptions['expectedChildContextValue']): ContextValueFilter | undefined {
+    return expectedChildContextValue ? { include: expectedChildContextValue } : undefined
+}
 
-    if (options?.expectedChildContextValue) {
-        return ext.appResourceTree.showTreeItemPicker(options.expectedChildContextValue, context, appResource);
-    } else {
-        return appResource as unknown as T;
+function convertAppResourceFilterToAzExtResourceType(filter?: PickAppResourceOptions['filter']): AzExtResourceType[] | undefined {
+    if (!filter) {
+        return undefined;
     }
+
+    filter = Array.isArray(filter) ? filter : [filter];
+
+    return filterMap(filter, (f) => getAzExtResourceType({ type: f.type, kind: f.kind }))
```

Comment:
Since `type` and `kind` are already properties on `f`, you might be able to do this:
```suggestion
    return filterMap(filter, (f) => getAzExtResourceType(f))
```

Reply:
Nice!